pkg/util/kubernetes/apiserver/wpa_controller.go

```diff
@@ -38,8 +55,69 @@ func (h *AutoscalersController) RunWPA(stopCh <-chan struct{}) {
 	<-stopCh
 }
 
-// ExtendToWPAController adds the handlers to the AutoscalersController to support WPAs
-func ExtendToWPAController(h *AutoscalersController, wpaInformer v1alpha1.WatermarkPodAutoscalerInformer) {
+type checkAPI func() error
+
+func tryCheckWPACRD(check checkAPI) error {
+	if err := check(); err != nil {
+		// Check if this is a known problem of missing CRD registration
+		if isWPACRDNotFoundError(err) {
+			return err
+		}
+		// In all other cases return a permanent error to prevent from retrying
+		log.Errorf("WPA CRD check failed: not retryable: %s", err)
+		return backoff.Permanent(err)
+	}
+	log.Info("WPA CRD check successful")
+	return nil
+}
+
+func notifyCheckWPACRD() backoff.Notify {
+	attempt := 0
+	return func(err error, delay time.Duration) {
+		attempt++
+		mins := int(delay.Minutes())
+		secs := int(math.Mod(delay.Seconds(), 60))
+		log.Warnf("WPA CRD missing (attempt=%d): will retry in %dm%ds", attempt, mins, secs)
+	}
+}
+
+func isWPACRDNotFoundError(err error) bool {
+	status, ok := err.(*apierrors.StatusError)
+	if !ok {
+		return false
+	}
+	reason := status.Status().Reason
+	details := status.Status().Details
+	return reason == v1.StatusReasonNotFound &&
+		details.Group == apis_v1alpha1.SchemeGroupVersion.Group &&
+		details.Kind == "watermarkpodautoscalers"
+}
+
+func checkWPACRD(wpaClient wpa_client.Interface) backoff.Operation {
+	check := func() error {
+		_, err := wpaClient.DatadoghqV1alpha1().WatermarkPodAutoscalers(v1.NamespaceAll).List(v1.ListOptions{})
```

Comment:
maybe you can use the `CustomResourceDefinition` client here.
something like:
```
client.ApiextensionsV1beta1().CustomResourceDefinitions().Get(...)
```

Reply:
Unfortunately the `clientSet` in WPA controller does not have `ApiextensionsV1beta()` API set.